src/ipc/utils/git_utils.ts

```diff
@@ -221,3 +270,285 @@ export async function getFileAtCommit({
     }
   }
 }
+
+export async function gitListBranches({
+  path,
+}: GitBaseParams): Promise<string[]> {
+  const settings = readSettings();
+
+  if (settings.enableNativeGit) {
+    const result = await exec(["branch", "--list"], path);
+
+    if (result.exitCode !== 0) {
+      throw new Error(result.stderr.toString());
+    }
+    // Parse output:
+    // e.g. "* main\n  feature/login"
+    return result.stdout
+      .toString()
+      .split("\n")
+      .map((line) => line.replace("*", "").trim())
+      .filter((line) => line.length > 0);
+  } else {
+    return await git.listBranches({
+      fs,
+      dir: path,
+    });
+  }
+}
+
+export async function gitRenameBranch({
+  path,
+  oldBranch,
+  newBranch,
+}: GitBranchRenameParams): Promise<void> {
+  const settings = readSettings();
+
+  if (settings.enableNativeGit) {
+    // git branch -m oldBranch newBranch
+    const result = await exec(["branch", "-m", oldBranch, newBranch], path);
+    if (result.exitCode !== 0) {
+      throw new Error(result.stderr.toString());
+    }
+  } else {
+    await git.renameBranch({
+      fs,
+      dir: path,
+      oldref: oldBranch,
+      ref: newBranch,
+    });
+  }
+}
+
+export async function gitClone({
+  path,
+  url,
+  accessToken,
+  singleBranch = true,
+  depth = 1,
```

Comment:
let's use `undefined` as the default value instead of  `1` to match git clone's behavior (seems more intuitive to me)